attack_mapper.py

```diff
@@ -0,0 +1,461 @@
+import json
+import os
+import re
+
+from typing import Dict, List, Any, Optional, Tuple
+
+# ----------------------------------------------------------------------------
+# Config
+# ----------------------------------------------------------------------------
+DEFAULT_URL = os.getenv(
+    "ATTACK_V18_URL",
+    "https://raw.githubusercontent.com/mitre-attack/attack-stix-data/master/enterprise-attack/enterprise-attack.json",
+)
+CACHE_PATH = os.getenv(
+    "ATTACK_V18_CACHE",
+    os.path.join(os.path.dirname(__file__), "uploads", "attack18_cache.json"),
+)
+
+_TID_RX = re.compile(r'(T\d{4}(?:\.\d{3})?)', re.IGNORECASE)
+
+
+def _extract_tids_from_analytic(a):
```

Comment:
looking at the analytics from the most recent ATT&CK v18 json bundle, Analytic names are always of the format "Analytic XXXX", and it looks like the external ID is also the ANXXXX ID. The descriptions seem to only mention technique IDs when including hyperlinks to other technique IDs, but those aren't always directly related to the analytic at hand. Description is a bit of a stretch, and we won't get techniques from the other fields. I'm considering just removing this method and sticking with explicit "detects" relationships to get analytics for a TID